tests/integration/test_bayesian_optimization.py

```diff
@@ -81,6 +81,7 @@
 # Optimizer parameters for testing against the branin function.
 # We also use these for a quicker test against a simple quadratic function
 # (regenerating is necessary as some of the acquisition rules are stateful).
+# The various   # type: ignore[arg-type]  are for rules that are only supported by GPR models.
```

Comment:
why do we need this?

Reply:
_test_optimizer_finds_minimum expects an acquisition_rule that can support any GPflowPredictor model, since it's sometimes run with VGP or SVGP models. However, some of the rules only support GPR.  At some point I might try to figure out how to refactor it without making the test look messier, but I think this is fine for now.